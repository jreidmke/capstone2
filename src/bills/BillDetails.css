#directions {
    padding-top: 20px;
}

#tables {
    padding-top: 60px;
}

#card, #title {
    padding-top: 2em;
}
