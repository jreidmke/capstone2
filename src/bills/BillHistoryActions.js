import Table from 'react-bootstrap/Table';


const BillHistoryActions = ({votes}) => {

    return(
        
        <Table striped bordered hover variant="light">
            <thead>
                <tr>
                    <td>Date</td>
                    <td>Chamber</td>
                    <td>Action Type</td>
                    <td>Description</td>
                </tr>
            </thead>
            
                <tbody>
                    {votes.map(v => (
                        <tr>
                            <td>{v.datetime}</td>
                            <td>{v.chamber}</td>
                            <td>{v.action_type}</td>
                            <td>{v.description}</td>
                        </tr>
                    ))}
                </tbody>
        </Table>
    )
}

export default BillHistoryActions;
