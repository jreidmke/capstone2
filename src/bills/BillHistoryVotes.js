import Table from 'react-bootstrap/Table';


const BillHistoryVotes = ({votes}) => {

    return(
        
        <Table striped bordered hover variant="light">
            <thead>
                <tr>
                    <td>Date</td>
                    <td>Question</td>
                    <td>Chamber</td>
                    <td>Yes Votes</td>
                    <td>No Votes</td>
                    <td>Not Voting</td>
                    <td>Result</td>
                </tr>
            </thead>
            
                <tbody>
                    {votes.map(v => (
                        <tr>
                            <td>{v.date}</td>
                            <td>{v.question}</td>
                            <td>{v.chamber}</td>
                            <td>{v.total_yes}</td>
                            <td>{v.total_no}</td>
                            <td>{v.total_not_voting}</td>
                            <td>{v.result}</td>
                        </tr>
                    ))}
                </tbody>
        </Table>
    )
}

export default BillHistoryVotes;