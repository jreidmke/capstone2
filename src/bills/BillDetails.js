import {useState, useEffect} from 'react';
import {useParams} from 'react-router-dom';
// import {getBillData} from './../api/api';
import {PropublicaApi} from './../api/api2';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import {FaArrowDown} from 'react-icons/fa';
import LoadingSpinner from './../common/LoadingSpinner';
import BillHistoryVotes from './BillHistoryVotes';
import BillHistoryActions from './BillHistoryActions'
import BillTitle from './BillTItle';
import BillCard from './../bill-card/BillCard'; 
import './BillDetails.css';
import {RiGovernmentFill, RiAwardFill} from 'react-icons/ri';
import {BsAwardFill} from 'react-icons/bs'
import {dateTimeFormatter} from './../helpers/helpers';

const BillDetails = () => {
    const {billId} = useParams();
    const [bill, setBill] = useState();
    const [isEnacted, setIsEnacted] = useState();

    useEffect(() => {
        async function getBill() {
            const resp = await PropublicaApi.getBillData(billId);
            setBill(resp);
            setIsEnacted(resp.enacted); 
        };
        getBill(); 
    }, []);


    return(
        <Container>

            <Row id='title'>
                {!bill ? <LoadingSpinner/> :
                <BillTitle bill={bill}/>}
            </Row>
            
            <Row>
                <Col/>
                <Col xs={6}>
                    <RiGovernmentFill size='10em'/>
                </Col>
                <Col>
                    {isEnacted ? 
                    <Col>
                        <h6><b>Bill Enacted on {dateTimeFormatter(isEnacted)}</b></h6>
                        <BsAwardFill size='10em' color='gold' id='enacted-seal'/>
                    </Col> : ""}
                </Col>
            </Row>

            <Row id='card'>
                <Col>
                    {bill ? <BillCard bill={bill}/> : <LoadingSpinner/>}
                </Col>
            </Row>

            <hr/>

            <Row id='directions'>
                <Col>
                    <h2>For More Information, Scroll Down</h2>
                </Col>
            </Row>

            <Row id='arrow'>
                <Col>
                    <FaArrowDown size='10em' color='white'/>
                </Col>
            </Row>

            <Row id='tables'>
                <Col>
                    <h3>Complete Bill Actions</h3>
                    {bill ? <BillHistoryActions votes={bill.actions}/> : ""}
                </Col>

                <Col>
                    <h3>Complete Bill Vote List</h3>
                    {bill ? <BillHistoryVotes votes={bill.votes}/> : ""}
                </Col>
            </Row>
            
        </Container>
    )
}

export default BillDetails;


