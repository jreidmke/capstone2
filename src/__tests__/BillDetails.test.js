import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from 'react-router-dom';
import BillDetails from './../bills/BillDetails';
import BillHistoryActions from './../bills/BillHistoryActions';
import BillHistoryVotes from './../bills/BillHistoryVotes';
import BillTitle from './../bills/BillTItle'; 
import billData from './../billdata.json'; 

//BILL DETAILS COMPONENT
test('renders without crashing', function() {
    const {getByText} = render(
        <MemoryRouter initialEntries={['/bills/hr501-116']}>
            <BillDetails/>
        </MemoryRouter>
    ); 
    expect(getByText('Complete Bill Actions')).toBeInTheDocument();
    expect(getByText('Complete Bill Vote List')).toBeInTheDocument();
}); 

test('it matches snapshot', function() {
    const {asFragment} = render(
        <MemoryRouter initialEntries={['/bills/hr501-116']}>
            <BillDetails/>
        </MemoryRouter>        
    );
    expect(asFragment()).toMatchSnapshot();
})

//IN HERE WE'LL TEST FOR BAD BILL NUMBERS WHICH MEANS 
//WE'LL NEED TO DO A LIL REDIRECTING BACK IN THE COMPONENT

/**BILL DETAILS CHILDREN COMPONENTS */

//BILL HISTORY ACTIONS
test('renders without crashing', function() {
    const {getByText} = render(

    )
})