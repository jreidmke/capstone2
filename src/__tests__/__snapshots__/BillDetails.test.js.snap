// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="row"
      id="title"
    >
      <div
        class="spinner-border"
        role="status"
      >
        <span
          class="sr-only"
        >
          Loading...
        </span>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      />
      <div
        class="col-6"
      >
        <svg
          fill="currentColor"
          height="10em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="10em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M2 19V8H1V6h3V4a1 1 0 0 1 1-1h14a1 1 0 0 1 1 1v2h3v2h-1v11h1v2H1v-2h1zm11 0v-7h-2v7h2zm-5 0v-7H6v7h2zm10 0v-7h-2v7h2zM6 5v1h12V5H6z"
            />
          </g>
        </svg>
      </div>
      <div
        class="col"
      />
    </div>
    <div
      class="row"
      id="card"
    >
      <div
        class="col"
      >
        <div
          class="spinner-border"
          role="status"
        >
          <span
            class="sr-only"
          >
            Loading...
          </span>
        </div>
      </div>
    </div>
    <hr />
    <div
      class="row"
      id="directions"
    >
      <div
        class="col"
      >
        <h2>
          For More Information, Scroll Down
        </h2>
      </div>
    </div>
    <div
      class="row"
      id="arrow"
    >
      <div
        class="col"
      >
        <svg
          color="white"
          fill="currentColor"
          height="10em"
          stroke="currentColor"
          stroke-width="0"
          style="color: white;"
          viewBox="0 0 448 512"
          width="10em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M413.1 222.5l22.2 22.2c9.4 9.4 9.4 24.6 0 33.9L241 473c-9.4 9.4-24.6 9.4-33.9 0L12.7 278.6c-9.4-9.4-9.4-24.6 0-33.9l22.2-22.2c9.5-9.5 25-9.3 34.3.4L184 343.4V56c0-13.3 10.7-24 24-24h32c13.3 0 24 10.7 24 24v287.4l114.8-120.5c9.3-9.8 24.8-10 34.3-.4z"
          />
        </svg>
      </div>
    </div>
    <div
      class="row"
      id="tables"
    >
      <div
        class="col"
      >
        <h3>
          Complete Bill Actions
        </h3>
      </div>
      <div
        class="col"
      >
        <h3>
          Complete Bill Vote List
        </h3>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
